src/pages/integrations/[unified-api]/[provider].tsx

```diff
@@ -34,21 +34,18 @@ const Connection = ({ token, jwt, connectionId }: IProps) => {
     })
   }
 
-  const { data, error } = useSWR('/vault/connections', fetcher, {
+  const { data, error } = useSWR(`/vault/connections/${unifiedApi}/${provider}`, fetcher, {
     shouldRetryOnError: false,
     revalidateOnFocus: false
   })
 
-  const connections: IConnection[] = data?.data?.data
+  const currentConnection: IConnection = data?.data?.data
 
   useEffect(() => {
-    const currentConnection = connections?.filter(
-      (connection: IConnection) => connection.id === connectionId
-    )[0]
     if (currentConnection) {
       setConnection(currentConnection)
     }
-  }, [connectionId, connections])
+  }, [currentConnection])
```

Comment:
Don't think we have to save the connection to the local state anymore, since we aren't making any changes to it.